src/coreComponents/constitutive/solid/SolidBase.hpp

```diff
@@ -121,11 +133,31 @@ class SolidBaseUpdates
    * @return the thermalExpansionCoefficient of element k
    */
   GEOS_HOST_DEVICE
-  real64 getThermalExpansionCoefficient( localIndex const k ) const
+  virtual real64 getThermalExpansionCoefficient( localIndex const k ) const
```

Comment:
why do you want this to be virtual? It's okay if you are just trying to enforce that function is implemented everywhere but keep in mind that there are no vtables on device. These functions can't be used as proper virtual functions.

Reply:
`virtual` is added to make it consistent with the existing getters of bulk and shear moduli.